cmd/snap/cmd_snap_op_test.go

```diff
@@ -302,6 +302,294 @@ func (s *SnapOpSuite) TestInstallUnaliased(c *check.C) {
+}, "kind": "snap-revision-not-available-for-architecture"}, "status-code": 404}`)
+	})
+
+	_, err := snap.Parser().ParseArgs([]string{"install", "foo"})
+	c.Assert(err, check.NotNil)
+	c.Check(fmt.Sprintf("\nerror: %v\n", err), check.Equals, `
+error: snap "foo" is not available on stable for this architecture (arm64) but
+       exists on other architectures (amd64).
+`)
+
+	c.Check(s.Stdout(), check.Equals, "")
+	c.Check(s.Stderr(), check.Equals, "")
+}
+
+func (s *SnapOpSuite) TestInstallSnapRevisionNotAvailableForArchitectureTrackAndRiskOther(c *check.C) {
+	s.RedirectClientToTestServer(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintln(w, `{"type": "error", "result": {"message": "no snap revision given constraints", "value": {
+  "snap-name": "foo",
+  "action": "install",
+  "architecture": "arm64",
+  "channel": "1.0/stable",
+  "releases": [{"architecture": "amd64", "channel": "stable"}]
+}, "kind": "snap-revision-not-available-for-architecture"}, "status-code": 404}`)
+	})
+
+	_, err := snap.Parser().ParseArgs([]string{"install", "--channel=1.0/stable", "foo"})
+	c.Assert(err, check.NotNil)
+	c.Check(fmt.Sprintf("\nerror: %v\n", err), check.Equals, `
+error: snap "foo" is not available on this architecture (arm64) but exists on
+       other architectures (amd64).
+`)
+
+	c.Check(s.Stdout(), check.Equals, "")
+	c.Check(s.Stderr(), check.Equals, "")
+}
+
+func (s *SnapOpSuite) TestInstallSnapRevisionNotAvailableInvalidChannel(c *check.C) {
+	s.RedirectClientToTestServer(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintln(w, `{"type": "error", "result": {"message": "no snap revision given constraints", "value": {
+  "snap-name": "foo",
+  "action": "install",
+  "architecture": "amd64",
+  "channel": "a/b/c/d",
+  "releases": [{"architecture": "amd64", "channel": "stable"}]
+}, "kind": "snap-revision-not-available-for-channel"}, "status-code": 404}`)
+	})
+
+	_, err := snap.Parser().ParseArgs([]string{"install", "--channel=a/b/c/d", "foo"})
+	c.Assert(err, check.NotNil)
+	c.Check(fmt.Sprintf("\nerror: %v\n", err), check.Equals, `
+error: requested what looks like an invalid channel for snap "foo". Please use
+       'snap info foo' to list available releases.
+`)
+
+	c.Check(s.Stdout(), check.Equals, "")
+	c.Check(s.Stderr(), check.Equals, "")
+}
+
+func (s *SnapOpSuite) TestInstallSnapRevisionNotAvailableForChannelNonExistingBranch(c *check.C) {
+	s.RedirectClientToTestServer(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintln(w, `{"type": "error", "result": {"message": "no snap revision given constraints", "value": {
+  "snap-name": "foo",
+  "action": "install",
+  "architecture": "amd64",
+  "channel": "stable/baz",
+  "releases": [{"architecture": "amd64", "channel": "stable"}]
+}, "kind": "snap-revision-not-available-for-channel"}, "status-code": 404}`)
+	})
+
+	_, err := snap.Parser().ParseArgs([]string{"install", "--channel=a/b/c/d", "foo"})
```

Comment:
Nitpick, but in this case the channel argument is not a valid branch name.

Reply:
yes, that's what the test is about, afaict we get a revision-not-found anyway, but we cannot try to match the requested channel to what is available  because the request channel is malformed, so we give an error here hinting at that

Reply:
What I meant, the channel param is an invalid channel name (which seems to be the scenario of the test above), and here you could have passed something like "--channel=stable/baz", which is a name that looks more like a branch (and what the expected error seems to indicate? I guess it is indifferent for the test, given the response is manually set, and that's why I said it was a nitpick :-)).

Reply:
ah, now I understand is about the value to --channel, fixing